storageClient/storageClient.go

```diff
@@ -74,12 +78,30 @@ func (c *StorageClient) Challenge(ctx context.Context, noteID string) ([]string,
 	return challenges, err
 }
 
+func (c *StorageClient) BulkDeleteByClient(ctx context.Context, clientID uuid.UUID, limit int) (BulkDeleteResponse, error) {
+	path := c.Host + "/internal" + storageServiceBasePath + "/notes/bulk/" + clientID.String()
+	request, err := e3dbClients.CreateRequest("DELETE", path, nil)
+	var resp BulkDeleteResponse
+	if err != nil {
+		return resp, err
+	}
+	urlParams := request.URL.Query()
+	if limit != 0 {
```

Comment:
do we need this check//do we need to also check for negative values?

Reply:
We could, the services check and deal with negative values. This check is really more of a way to imply that you want to not attach the query param by putting zero. The other way to do that would be a pointer to the int and check for nil? pointers to ints just feels weird to me, but might be more appropriate.